pkg/mig/reconfigure/clients.go

```diff
@@ -0,0 +1,344 @@
+			LabelSelector: fmt.Sprintf("app=%s", o.app),
+		},
+	)
+	if err != nil {
+		return fmt.Errorf("unable to delete pods for app %s: %w", o.app, err)
+	}
+	return nil
+}
+
+func (o *pod) waitForDeletion() error {
+	timeout := 5 * time.Minute
+	ctx, cancel := context.WithTimeout(context.Background(), timeout)
+	defer cancel()
+
+	watcher, err := o.node.clientset.CoreV1().Pods(o.namespace).Watch(ctx, metav1.ListOptions{
+		FieldSelector: fmt.Sprintf("spec.nodeName=%s", o.node.name),
+		LabelSelector: fmt.Sprintf("app=%s", o.app),
+	})
+	if err != nil {
+		return fmt.Errorf("unable to watch pods for deletion: %w", err)
+	}
+	defer watcher.Stop()
+
+	// Wait for all matching pods to be deleted
+	for {
+		select {
+		case <-ctx.Done():
+			return fmt.Errorf("timeout waiting for pod deletion: %w", ctx.Err())
+		case event := <-watcher.ResultChan():
+			if event.Type == watch.Deleted {
+				// Check if there are any remaining pods matching our criteria
+				pods, err := o.node.clientset.CoreV1().Pods(o.namespace).List(ctx, metav1.ListOptions{
+					FieldSelector: fmt.Sprintf("spec.nodeName=%s", o.node.name),
+					LabelSelector: fmt.Sprintf("app=%s", o.app),
+				})
+				if err != nil {
+					return fmt.Errorf("unable to list pods: %w", err)
+				}
+				if len(pods.Items) == 0 {
+					// All pods have been deleted
+					break
+				}
+			}
+		}
+	}
+}
+
+// Restart the specified operand by setting its deployLabel to 'true'
+// If the deploy label is already set to false, this is assumed to be controlled
+// by an external entity and no changes are made.
+func (o *operand) Restart() error {
+	if o.lastValue == "false" {
+		return nil
+	}
+	err := o.node.setNodeLabelValue(o.deployLabel, "true")
+	if err != nil {
+		return fmt.Errorf("unable to restart operand %q: %w", o.app, err)
+	}
+	return nil
+}
+
+// Stop the specified operand by setting its deploy label to 'paused-for-mig-change'.
+// If the deploy label is already set to false, this is assumed to be controlled
+// by an external entity and no changes are made.
+func (o *operand) Stop() error {
+	value, err := o.node.getNodeLabelValue(o.deployLabel)
+	if err != nil {
+		return fmt.Errorf("unable to get the value of the %q label: %w", o.deployLabel, err)
+	}
+	o.lastValue = value
+	if value != "false" {
+		if err := o.node.setNodeLabelValue(o.deployLabel, "paused-for-mig-change"); err != nil {
+			return err
+		}
+	}
+	// TODO: For the nvidia.com/gpu.deploy.nvsm label we have no associated app name.
+	if o.app == "" {
+		return nil
+	}
+	return o.waitForDeletion()
+}
+
+func (opts *reconfigureMIGOptions) newSystemdService(name string) *systemdService {
+	return &systemdService{
+		name:          name,
+		hostRootMount: opts.HostRootMount,
+	}
+}
+
+func (o *systemdService) Restart() error {
+	cmd := exec.Command("chroot", o.hostRootMount, "systemctl", "start", o.name) // #nosec G204 -- HostRootMount validated via dirpath, service validated via systemd_service_name.
```

Comment:
Can we use a go library instead of shelling out?

Would this work? https://github.com/coreos/go-systemd